### PR TITLE
feat; Adds a new feature - Define device model

### PR DESCRIPTION
**Pull request by ChetanaRK**
@NagarajNidagundi, @AmitkumarEvry and @sinaj-evry 
Please review the Device model set up and let me know for any changes.

### CODE REVIEW

File: src/WWTelematics.DeviceService/Models/DeviceContext.cs [hunk 1]
```diff
@@ -3,32 +3,31 @@
...
         /// <param name="options">Options for the DBContext.</param>
-        public YourDbContext(DbContextOptions<YourDbContext> options)
+        public DeviceContext(DbContextOptions<DeviceContext> options)
             : base(options)
         {
         }
 
         /// <summary>
-        /// Gets or sets the YourEntity table.
+        /// Gets or sets the device service table.
```
Comment (AmitkumarEvry) [inline src/WWTelematics.DeviceService/Models/DeviceContext.cs]: change text to "Gets or sets the device table."

File: src/WWTelematics.DeviceService/Models/DeviceContext.cs [hunk 2]
```diff
@@ -52,20 +51,28 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                    .HasColumnName("label")
+                    .HasMaxLength(64).IsRequired();
+
+                device.Property(e => e.TenantId)
+                .HasColumnName("tenant_id").IsRequired();
+
+                device.Property(e => e.CreatedTime)
+                   .HasColumnName("created_time").IsRequired();
+
+                device.Property(e => e.Active);
```
Comment (AmitkumarEvry) [inline src/WWTelematics.DeviceService/Models/DeviceContext.cs]: Add .HasColumnName("active") here

Comment (ChetanaRK) [reply src/WWTelematics.DeviceService/Models/DeviceContext.cs]: Done.

File: src/WWTelematics.DeviceService/Models/DeviceContext.cs (hunk 1 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService/Models/DeviceContext.cs]: Done.

File: src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                .HasAnnotation("ProductVersion", "2.2.6-servicing-10079")
+                .HasAnnotation("Relational:MaxIdentifierLength", 63);
+
+            modelBuilder.Entity("WWTelematics.DeviceService.Models.DeviceModel", b =>
+                {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd()
+                        .HasColumnName("id");
+
+                    b.Property<bool>("Active");
```
Comment (NagarajNidagundi) [inline src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs]: In all places, Coumn Name should be small letter "active",

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    /// <summary>
+    /// Device Model.
+    /// </summary>
+    public class DeviceModel
+    {
+        /// <summary>
+        /// Gets the id of the entity.
+        /// </summary>
+        [JsonIgnore]
+        public int Id { get; private set; }
```
Comment (AmitkumarEvry) [inline src/WWTelematics.DeviceService/Models/DeviceModel.cs]: Remove private

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        /// <summary>
+        /// Gets or sets a label of the entity.
+        /// </summary>
+        [JsonProperty(PropertyName = "label")]
+        public string Label { get; set; }
+
+        /// <summary>
+        /// Gets or sets a created_time of the entity.
+        /// </summary>
+        [JsonProperty(PropertyName = "created_time")]
```
Comment (AmitkumarEvry) [inline src/WWTelematics.DeviceService/Models/DeviceModel.cs]: Use camelCase for all json properties. rename this to "createdTime"

File: src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs (hunk 3 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs]: Its a property name. Column name genrated is "active" only.

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs (hunk 4 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService/Models/DeviceModel.cs]: Removed.

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs (hunk 5 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService/Models/DeviceModel.cs]: Changed.

File: src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs (hunk 3 above)
Comment (AmitkumarEvry) [reply src/WWTelematics.DeviceService.Migrations/Migrations/DeviceContextModelSnapshot.cs]: Chetana is right here.

**Comment by AmitkumarEvry** [comment]
Looks good

**Comment by sinaj-evry** [comment]
looks good

**Comment by ChetanaRK** [comment]
@davidw-foxtrax, kindly review the define device model user story.

File: src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj [hunk 6]
```diff
@@ -15,4 +15,8 @@
     <ProjectReference Include="..\WWTelematics.DeviceService\WWTelematics.DeviceService.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (davidw-foxtrax) [inline src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj]: What is the benefit of including this ItemGroup?

(We did not include this in Client.)

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace WWTelematics.DeviceService.Models
+{
+    /// <summary>
+    /// Device Model.
+    /// </summary>
+    public class DeviceModel
```
Comment (davidw-foxtrax) [inline src/WWTelematics.DeviceService/Models/DeviceModel.cs]: The name of this class should just be "Device"

File: src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj (hunk 6 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj]: This was automatically added when I created DB. As we didn't had folder created in master branch for Migrations, it is auto generated.

File: src/WWTelematics.DeviceService/Models/DeviceModel.cs (hunk 7 above)
Comment (ChetanaRK) [reply src/WWTelematics.DeviceService/Models/DeviceModel.cs]: Changed as suggested.

File: src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj (hunk 6 above)
Comment (davidw-foxtrax) [reply src/WWTelematics.DeviceService.Migrations/WWTelematics.DeviceService.Migrations.csproj]: I've removed this ItemGroup.

Do you recall which command it was that generated this line?  I'm curious why it generated here but not in the other repos.  If it is something that is needed, we ought to add it everywhere.

**Pull request merged**